solidity/ecdsa/test/WalletRegistry.WalletCreation.test.ts

```diff
@@ -0,0 +1,3238 @@
+          expectedMembers.splice(26, 1) // index -4
+          expectedMembers.splice(55, 1) // index -5
+          expectedMembers.splice(58, 1) // index -6
+
+          expect(
+            (await walletRegistry.getWallet(walletID)).membersIdsHash
+          ).to.be.equal(hashUint32Array(expectedMembers))
+        })
+
+        // it("should ban misbehaved operators from sortition pool rewards", async () => {
+        //   const now = await helpers.time.lastBlockTime()
+        //   const expectedUntil = now + params.sortitionPoolRewardsBanDuration
+
+        //   await expect(tx)
+        //     .to.emit(sortitionPool, "IneligibleForRewards")
+        //     .withArgs(misbehavedIds, expectedUntil)
+        // })
+
+        it("should clean dkg data", async () => {
+          await assertDkgResultCleanData(walletRegistry)
+        })
+      })
+
+      // This case shouldn't happen in real life. When a result is submitted
+      // with invalid order of misbehaved operators it should be challenged.
+      context(
+        "when misbehaved operators are not in ascending order",
+        async () => {
+          const misbehavedIndices = [2, 9, 30, 11, 60, 64]
+
+          let dkgResult: DkgResult
+          let submitter: SignerWithAddress
+
+          before(async () => {
+            await createSnapshot()
+
+            await mineBlocksTo(startBlock + constants.offchainDkgTime)
+            ;({ dkgResult, submitter } = await signAndSubmitCorrectDkgResult(
+              walletRegistry,
+              groupPublicKey,
+              dkgSeed,
+              startBlock,
+              misbehavedIndices
+            ))
+
+            await mineBlocks(params.dkgResultChallengePeriodLength)
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          it("should succeed", async () => {
+            await expect(
+              walletRegistry.connect(submitter).approveDkgResult(dkgResult)
+            ).to.not.be.reverted
+          })
+        }
+      )
+
+      context("when misbehaved members contains duplicates", async () => {
+        const misbehavedIndices = [2, 9, 9, 10]
+
+        let dkgResult: DkgResult
+        let submitter: SignerWithAddress
+
+        before(async () => {
+          await createSnapshot()
+
+          await mineBlocksTo(startBlock + constants.offchainDkgTime)
+          ;({ dkgResult, submitter } = await signAndSubmitCorrectDkgResult(
+            walletRegistry,
+            groupPublicKey,
+            dkgSeed,
+            startBlock,
+            misbehavedIndices
+          ))
+
+          await mineBlocks(params.dkgResultChallengePeriodLength)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should succeed", async () => {
+          await expect(
+            walletRegistry.connect(submitter).approveDkgResult(dkgResult)
+          ).to.not.be.reverted
+        })
+      })
+    })
+
+    // context(
```

Comment:
I'm kind of on the fence when it comes to committing commented out tests. Just wanted to bring your attention, but if you think it's okay since we will deal with the rewards later, I guess we can leave it.

Reply:
Yeah, I'm not sure about it as well. I imagined it will be helpful when adding the other functionalities to see what parts were copied from the original implementation and just uncomment them.

Reply:
ok, lets leave it and cleanup in the next prs